pkg/network/tracer/tracer_test.go

```diff
@@ -1017,13 +1017,16 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
 	queryMsg.RecursionDesired = true
 
 	dnsClient := new(dns.Client)
-	dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-	require.NoError(t, err)
-	defer dnsConn.Close()
-	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
-	require.Eventually(t, func() bool {
+	var dnsClientAddr *net.UDPAddr
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
```

Comment:
This seems susceptible to the same problem I've mentioned in other DNS tests where timeouts are overcounted.

Reply:
yea... I was thinking of removing these retries entirely and just increasing the tracer timeout for now

Reply:
so actually we only log a timeout in the tracer when we get a response... in the failure case for this test we don't, and shouldn't see a timeout